### PR TITLE
[Scripts] Add Toybox AC Scripts, cleanup their headers, organize

### PR DESCRIPTION
**Pull request by imgntn**
This PR adds all of the Toybox AC scripts.  They're mostly used for ambient sound, but also one for a flickering light, and one to reset the basketball and target positions.

### CODE REVIEW

File: examples/toys/AC_scripts/fireplace_sound.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+var injector = null;
+var count = 300;
+
+Script.update.connect(function() {
+  if (count > 0) {
+    count--;
+    return;
+  }
+  
+  if (sound.downloaded && injector === null) {
```
Comment (ZappoMan) [inline examples/toys/AC_scripts/fireplace_sound.js]: maybe we should convert this script to use the same "checkDownloaded" strategy as `dogs_barking_sound.js`

```
// Check once a second to see if the audio file has been downloaded
var checkDownloadedTimer = Script.setInterval(checkDownloaded, 1000);
```

File: examples/toys/AC_scripts/cat_purr_sound.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+var injector = null;
+var count = 300;
+
+Script.update.connect(function() {
+  if (count > 0) {
+    count--;
+    return;
+  }
+  
+  if (sound.downloaded && injector === null) {
```
Comment (ZappoMan) [inline examples/toys/AC_scripts/cat_purr_sound.js]: maybe we should convert this script to use the same "checkDownloaded" strategy as `dogs_barking_sound.js`

```
// Check once a second to see if the audio file has been downloaded
var checkDownloadedTimer = Script.setInterval(checkDownloaded, 1000);
```

File: examples/toys/AC_scripts/insects_sound.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+var injector = null;
+var count = 300;
+
+Script.update.connect(function() {
+  if (count > 0) {
+    count--;
+    return;
+  }
+  
+  if (sound.downloaded && injector === null) {
```
Comment (ZappoMan) [inline examples/toys/AC_scripts/insects_sound.js]: maybe we should convert this script to use the same "checkDownloaded" strategy as `dogs_barking_sound.js`

```
// Check once a second to see if the audio file has been downloaded
var checkDownloadedTimer = Script.setInterval(checkDownloaded, 1000);
```

File: examples/toys/AC_scripts/river_water_sound.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+var injector = null;
+var count = 300;
+
+Script.update.connect(function() {
+  if (count > 0) {
+    count--;
+    return;
+  }
+  
+  if (sound.downloaded && injector === null) {
```
Comment (ZappoMan) [inline examples/toys/AC_scripts/river_water_sound.js]: maybe we should convert this script to use the same "checkDownloaded" strategy as `dogs_barking_sound.js`

```
// Check once a second to see if the audio file has been downloaded
var checkDownloadedTimer = Script.setInterval(checkDownloaded, 1000);
```

File: examples/toys/AC_scripts/windmill_sound.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+var injector = null;
+var count = 300;
+
+Script.update.connect(function() {
+  if (count > 0) {
+    count--;
+    return;
+  }
+  
+  if (sound.downloaded && injector === null) {
```
Comment (ZappoMan) [inline examples/toys/AC_scripts/windmill_sound.js]: maybe we should convert this script to use the same "checkDownloaded" strategy as `dogs_barking_sound.js`

```
// Check once a second to see if the audio file has been downloaded
var checkDownloadedTimer = Script.setInterval(checkDownloaded, 1000);
```

**Comment by ZappoMan** [comment]
why are we moving these to "toys" shouldn't they be in `./examples/acScripts"??

**Comment by imgntn** [comment]
@ZappoMan they're all specific to toybox. which is what /toys is.  if you want to run a toybox, you need the toys, the AC scripts, and the .svo, which i'm not sure we're ready to provide because of something about the models and licensing(?)  anyhow, better to keep it all in one place.  generally, i don't find AC/entity/client script to be a useful distinction at the top level

**Comment by imgntn** [comment]
i am going to combine all of the sound scripts into a single script, since they're all doing essentially the same thing

**Comment by imgntn** [comment]
they're combined now.  test with  https://rawgit.com/imgntn/hifi/toybox_ac/examples/toys/AC_scripts/toybox_sounds.js

**Pull request merged**